src/fluidsynth.c

```diff
@@ -1181,8 +1195,4 @@ print_help(fluid_settings_t *settings)
     {
         FLUID_FREE(midi_options);
     }
-
-    delete_fluid_settings(settings);
```

Comment:
Removing that call leaks the settings object.

Reply:
Well, there are plenty of places where we simply `exit(0)` without any cleanup. And freeing `settings` in a function called `print_settings()` just feels plain wrong to me.

Reply:
> Well, there are plenty of places where we simply exit(0) without any cleanup.

Probably, but you should not produce more leaks ;)

> And freeing settings in a function called print_settings() just feels plain wrong to me.

I understand, then just place it in the switch-case :)

Reply:
But does fluidsynth run on any platform where an `exit()` does not free all allocated memory anyway?

Reply:
> But does fluidsynth run on any platform where an exit() (or rather the OS after the process exists) does not free all allocated memory anyway?

I don't like this kind of arguing. I don't believe in that kind of "Alles-wird-wieder-gut" process that collects our resource garbage. This is our job. But since you've asked for a specific OS, I'd suggest Windows95 (Carlo has tested it recently).

Reply:
Ah, didn't know that Win95 does not use process-based virtual memory. Ok, I get your point (and personally I like cleanup up better as well).